### PR TITLE
Update JavaTemplate recipe guide for 8.0

### PR DESCRIPTION
**Pull request by mike-solomon**

### CODE REVIEW

File: authoring-recipes/modifying-methods-with-javatemplate.md [hunk 1]
```diff
@@ -257,12 +257,12 @@ As mentioned in our [best practice guide](recipe-conventions-and-best-practices.
...
 
 ```java
-protected JavaIsoVisitor<ExecutionContext> getVisitor() {
+public JavaIsoVisitor<ExecutionContext> getVisitor() {
     return new JavaIsoVisitor<ExecutionContext>() {
         // ...
 
         // Template used to insert two additional parameters into the "setCustomerInfo()" method declaration
-        private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder(this::getCursor, "Date dateOfBirth, String firstName, #{}")
+        private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder( "Date dateOfBirth, String firstName, #{}")
```
Comment (jsoref) [inline authoring-recipes/modifying-methods-with-javatemplate.md:265]:
```suggestion
        private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder("Date dateOfBirth, String firstName, #{}")
```

File: authoring-recipes/modifying-methods-with-javatemplate.md [hunk 2]
```diff
@@ -275,15 +275,17 @@ protected JavaIsoVisitor<ExecutionContext> getVisitor() {
...
```
Comment (jsoref) [inline authoring-recipes/modifying-methods-with-javatemplate.md:288]:
```suggestion
        private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder("Date dateOfBirth, String firstName, #{}")
```

File: authoring-recipes/modifying-methods-with-javatemplate.md [hunk 3]
```diff
@@ -348,26 +355,25 @@ public void setCustomerInfo(Date dateOfBirth, String firstName, String lastName)
...
 
 ```java
-protected JavaIsoVisitor<ExecutionContext> getVisitor() {
+public JavaIsoVisitor<ExecutionContext> getVisitor() {
     return new JavaIsoVisitor<ExecutionContext>() {
         // ...
 
         // Template used to add a method body to "setCustomerInfo()" method declaration
-        private final JavaTemplate addMethodBodyTemplate = JavaTemplate.builder(this::getCursor, "").build();
+        private final JavaTemplate addMethodBodyTemplate = JavaTemplate.builder( " ").build();
```
Comment (jsoref) [inline authoring-recipes/modifying-methods-with-javatemplate.md:368]:
```suggestion
        private final JavaTemplate addMethodBodyTemplate = JavaTemplate.builder(" ").build();
```

File: authoring-recipes/modifying-methods-with-javatemplate.md [hunk 4]
```diff
@@ -462,24 +474,26 @@ public class ExpandCustomerInfo extends Recipe {
...
     @Override
-    protected JavaIsoVisitor<ExecutionContext> getVisitor() {
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
         return new JavaIsoVisitor<ExecutionContext>() {
             // Used to identify the method declaration that will be refactored
             private final MethodMatcher methodMatcher = new MethodMatcher("com.yourorg.Customer setCustomerInfo(String)");
 
             // Template used to insert two additional parameters into the "setCustomerInfo()" method declaration
-            private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder(this::getCursor, "Date dateOfBirth, String firstName, #{}")
+            private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder( "Date dateOfBirth, String firstName, #{}")
```
Comment (jsoref) [inline authoring-recipes/modifying-methods-with-javatemplate.md:483]:
```suggestion
            private final JavaTemplate addMethodParametersTemplate = JavaTemplate.builder("Date dateOfBirth, String firstName, #{}")
```

File: authoring-recipes/modifying-methods-with-javatemplate.md [hunk 5]
```diff
@@ -275,15 +275,17 @@ protected JavaIsoVisitor<ExecutionContext> getVisitor() {
...
 {% endhint %}
 
-With that template defined, we can now use it to replace the existing parameters via the `withTemplate()` function. This function takes in a `JavaTemplate`, some coordinates (where our template should be applied), and then 0 or more optional parameters (which will replace the interpolation marker we specified earlier):
+With that template defined, we can now use it to replace the existing parameters via the `apply()` function. The `apply` function expects a cursor, some coordinates (where our template should be applied), and then 0 or more optional parameters (which will replace the interpolation marker we specified earlier).
+
+The cursor parameter will either need to be `new Cursor(...)` or `updateCursor(...)` depending on whether or not the visitor has changed the `J` class anywhere. In our case, we removed the `abstract` modifier and will continue to make changes to where the cursor currently is – so we will need to update the cursor accordingly rather than making a new one.
```
Comment (kunli2) [inline authoring-recipes/modifying-methods-with-javatemplate.md:280]: I think `getCursor()` is the most common here.

**Review by kunli2 — APPROVED** [review]

**Pull request merged**